spec/models/photo_spec.rb

```diff
@@ -141,36 +141,43 @@ def with_carrierwave_processing(&block)
   end
 
   context 'with a saved photo containing EXIF data' do
-    before do
-      @exif_filename  = 'exif.jpg'
-      @exif_name      = File.join(File.dirname(__FILE__), '..', 'fixtures', @exif_filename)
-    end
 
-    it 'should contain EXIF data if user prefer' do
-      @alice_photo  = alice.build_post(:photo, :user_file => File.open(@exif_name), :to => alice.aspects.first.id)
+    let (:base_path)       { File.dirname(__FILE__) }
+    let (:public_path)     { File.join(base_path, '../../public/') }
+    let (:photo_with_exif) { File.open(File.join(base_path, '..', 'fixtures', 'exif.jpg')) }
 
-      with_carrierwave_processing do
-        @alice_photo.unprocessed_image.store! File.open(@exif_name)
-        @alice_photo.save
-      end
+    after do
+      FileUtils.rm_r Dir.glob(File.join(public_path, 'uploads/images/*'))
+    end
+
+    it 'should preserve EXIF data in according to user preference' do
+      image = image_from a_photo_sent_by(alice)
 
-      new_filename = File.join(File.dirname(__FILE__), '../../public/', @alice_photo.unprocessed_image.store_dir, @alice_photo.unprocessed_image.filename)
-      image = MiniMagick::Image.new(new_filename)
       expect(image.exif.length).not_to eq(0)
     end
 
-    it 'should not contain EXIF data if user prefer' do
-      @bob_photo  = bob.build_post(:photo, :user_file => File.open(@exif_name), :to => @aspect.id)
+    it 'should not preserve EXIF in according to user preference' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.